src/util/mod.rs

```diff
@@ -452,6 +452,24 @@ impl<T> Drop for MustConsumeVec<T> {
     }
 }
 
+/// Unknown enum values are stored in unknown fields since rust-protobuf
+/// 1.5.0, instead of reporting error when parse the message. So we must
+/// check the unknown fields if the enum's default value isn't invalid.
+///
+/// `pb_unknown_enum` returns true if the message's unknown fields contain
+/// an enum with id `field_id`. `field_id` for a given enum can be checked
+/// from proto files.
+///
+/// This should be removed after `ProtobufEnumOrUnknown`
+/// [implemented]https://github.com/stepancheg/rust-protobuf/issues/233).
+pub fn pb_unknown_enum<M: Message>(m: &M, field_id: u32) -> bool {
```

Comment:
how can we meet this problem? do we have any unknown enum?

Do you want to fix a compatibility problem here?

Reply:
The problem is if we use latest rust-protobuf, some logic like this:
```rust
if enum_from_message == EnumType::default_value {
    // do some logic
}
```
will be not trusted if the messages is from new version `Enum`.

Details is [here](https://github.com/stepancheg/rust-protobuf/issues/233).

Here what I want to do is make the `default check` more carefully.